### PR TITLE
Add ability to modify norwegian insurances

### PR DESCRIPTION
**Pull request by vonElfvin**
- Adds Norwegian quotues to all dtos and their creators

### CODE REVIEW

File: src/main/java/com/hedvig/backoffice/services/underwriter/dtos/QuoteRequestDto.kt [hunk 1]
```diff
@@ -11,9 +10,11 @@ data class QuoteRequestDto(
   val birthDate: LocalDate?,
   val ssn: String?,
   val quotingPartner: String?,
-  val productType: com.hedvig.backoffice.services.underwriter.dtos.ProductType?,
+  val productType: ProductType?,
```
Comment (AlexSJ1990) [inline src/main/java/com/hedvig/backoffice/services/underwriter/dtos/QuoteRequestDto.kt:13]: why did you change the productType?

Comment (vonElfvin) [reply src/main/java/com/hedvig/backoffice/services/underwriter/dtos/QuoteRequestDto.kt:13]: I guess I only updated the import not the type itself

Comment (vonElfvin) [reply src/main/java/com/hedvig/backoffice/services/underwriter/dtos/QuoteRequestDto.kt:13]: And if I did I only removed types like "SUBLETED" and stuff that is never used anywhere and shouldn't have been there is the first place

File: src/main/java/com/hedvig/backoffice/graphql/types/Quote.kt [hunk 2]
```diff
@@ -59,8 +63,43 @@ data class Quote(
         memberId = quote.memberId,
         startDate = quote.startDate,
         price = quote.price,
-        data = when {
-          quote.data is com.hedvig.backoffice.services.underwriter.dtos.QuoteData.ApartmentQuoteData -> QuoteData.ApartmentQuoteData(
+        data = when (quote.data) {
+            is com.hedvig.backoffice.services.underwriter.dtos.QuoteData.ApartmentData -> QuoteData.ApartmentQuoteData(
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/backoffice/graphql/types/Quote.kt:67]: Super minor: I like using typealias if it has the same name as another class, removing `com.hedvig.backoffice.services.underwriter.dtos.` is nice

File: src/main/java/com/hedvig/backoffice/graphql/GraphQLMutation.java [hunk 3]
```diff
@@ -414,11 +415,30 @@ public Quote activateQuote(final UUID id, final LocalDate activationDate, @Nulla
...
     return Quote.from(underwriterService.getQuote(createdQuoteId));
   }
 
+  public Quote createQuoteFromAgreement(
+    final UUID agreementId,
+    final String memberId,
+    final DataFetchingEnvironment env
+  ) {
+    final UUID createQuoteId = underwriterService.createQuoteFromAgreement(
+      new QuoteRequestFromAgreementDto(
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/graphql/GraphQLMutation.java:433]:
```suggestion
      new QuoteFromAgreementRequestDto(
```

File: src/main/java/com/hedvig/backoffice/services/underwriter/UnderwriterClient.kt [hunk 4]
```diff
@@ -40,9 +36,17 @@ interface UnderwriterClient {
...
+  @PostMapping("/_/v1/quotes/add/agreement")
+  fun addAgreementFromQuote(@RequestBody request: AddAgreementFromQuoteRequest): QuoteDto
+
   @GetMapping("/_/v1/quotes/members/{memberId}")
   fun getQuotes(@PathVariable("memberId") memberId: String): List<QuoteDto>
 
   @GetMapping("/_/v1/quotes/{quoteId}")
   fun getQuote(@PathVariable("quoteId") quoteId: UUID): QuoteDto
+
+  @PostMapping("/_/v1/quotes/createQuoteFromAgreement")
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/services/underwriter/UnderwriterClient.kt:48]: Very subjective but I like endpoints that look like this
```suggestion
  @PostMapping("/_/v1/quotes/create/from/agreement")
```

File: src/main/java/com/hedvig/backoffice/services/underwriter/UnderwriterServiceImpl.kt [hunk 5]
```diff
@@ -63,6 +57,17 @@ class UnderwriterServiceImpl(
     }
   }
 
+  private fun underwriterCreateQuoteFromAgreement(
+    quoteRequest: QuoteRequestFromAgreementDto
+  ): QuoteResponseDto? {
+    return try {
+      underwriterClient.createQuoteFromAgreement(quoteRequest)
+    } catch (e: ExternalServiceBadRequestException) {
+      logger.error("Failed to complete quote", e)
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/backoffice/services/underwriter/UnderwriterServiceImpl.kt:66]:
```suggestion
      logger.error("Failed to create quote", e)
```

File: src/main/resources/schema.graphqls [hunk 6]
```diff
@@ -56,16 +56,27 @@ type MutationType {
   setClaimFileCategory(claimId: ID!, claimFileId: ID!, category: String): String
   addInventoryItem(item: InventoryItemInput!): Boolean
   removeInventoryItem(inventoryItemId: ID!): Boolean
-  activateQuote(
-    id: ID!
-    activationDate: LocalDate!
+  activateQuote(id: ID!,
```
Comment (vonElfvin) [inline src/main/resources/schema.graphqls:59]:
```suggestion
  activateQuote(
    id: ID!,
```

File: src/main/resources/schema.graphqls [hunk 7]
```diff
@@ -56,16 +56,27 @@ type MutationType {
...
+    activationDate: LocalDate!,
     terminationDate: LocalDate
   ): Quote!
+
+  addAgreementFromQuote(id: ID!,
```
Comment (vonElfvin) [inline src/main/resources/schema.graphqls:64]:
```suggestion
  addAgreementFromQuote(
    id: ID!,
```

File: src/main/java/com/hedvig/backoffice/services/underwriter/dtos/QuoteDto.kt [hunk 8]
```diff
@@ -11,129 +13,224 @@ import java.time.LocalDate
...
+  RENT,
+  STUDENT_BRF,
+  STUDENT_RENT
+}
+
+enum class NorwegianHomeContentType {
+  RENT,
+  OWN,
+  STUDENT_RENT,
+  STUDENT_OWN
```
Comment (FredrikAreschoug) [inline src/main/java/com/hedvig/backoffice/services/underwriter/dtos/QuoteDto.kt:69]: Just realized that it should probably be `YOUTH_RENT` and `YOUTH_OWN` for norway
